Gems/O3DEMaterialEditor/Code/Source/O3DEMaterialEditorWidget.cpp

```diff
@@ -0,0 +1,57 @@
+
+#include <AzCore/Utils/Utils.h>
+
+#include <QLabel>
+#include <QVBoxLayout>
+
+#include <O3DEMaterialEditorWidget.h>
+#include <O3DEMaterialEditorSystemComponent.h>
+
+namespace O3DEMaterialEditor
+{
+    O3DEMaterialEditorWidget::O3DEMaterialEditorWidget(QWidget* parent)
+        : AzQtComponents::TabWidget(parent)
+    {
+        AzQtComponents::TabWidget::applySecondaryStyle(this, false/*bordered*/);
+
+        setMovable(true);
+
+        if (auto* o3deMaterialEditorSystem = GetO3DEMaterialEditorSystem())
+        {
+            const auto& registeredTabs = o3deMaterialEditorSystem->GetRegisteredTabs();
+            for (const auto& registeredTab : registeredTabs)
+            {
+                const AZStd::string& tabName = registeredTab.first;
+                const WidgetCreationFunc& widgetCreationFunc = registeredTab.second;
+
+                // Have render material tab first
+                if (tabName == "Render Materials")
+                {
+                    insertTab(0, widgetCreationFunc(nullptr), tabName.c_str());
+                    setCurrentIndex(0);
+                }
+                else
+                {
+                    addTab(widgetCreationFunc(nullptr), tabName.c_str());
+                }
+            }
+        }
```

Comment:
This is the O3DE Material Editor tab widget, which it's getting all the registered tabs and adding them to itself.